FindOnPage.md

```diff
@@ -0,0 +1,664 @@
+  COREWEBVIEW2_FIND_DIRECTION_BACKWARD,
+} COREWEBVIEW2_FIND_DIRECTION;
+
+
+/// Interface that provides methods related to the environment settings of CoreWebView2.
+/// This interface allows for the creation of new find configuration objects.
+// MSOWNERS: core (maxwellmyers@microsoft.com)
+[uuid(f10bddd3-bb59-5d5b-8748-8a1a53f65d0c), object, pointer_default(unique)]
+interface ICoreWebView2StagingEnvironment5 : IUnknown {
+  /// Creates a new instance of a FindConfiguration object.
+  /// This configuration object can be used to define parameters for a find operation.
+  /// Returns the newly created FindConfiguration object.
+  // MSOWNERS: core (maxwellmyers@microsoft.com)
+  HRESULT CreateFindConfiguration(
+      [out, retval] ICoreWebView2StagingFindConfiguration** value
+  );
+
+
+}
+
+
+/// Receives the result of the `StartFind` method.
+[uuid(7c20f8b0-c14e-5135-a099-6c9d11d59dd1), object, pointer_default(unique)]
+interface ICoreWebView2StagingFindStartFindCompletedHandler : IUnknown {
+
+  /// Provides the result of the corresponding asynchronous method.
+  HRESULT Invoke([in] HRESULT errorCode);
+}
+
+/// Receives `FindActiveMatchIndexChanged` events.
+[uuid(8d3422bf-66df-5bae-9916-71fd23d5bef6), object, pointer_default(unique)]
+interface ICoreWebView2StagingFindActiveMatchIndexChangedEventHandler : IUnknown {
+  /// Provides the event args for the corresponding event.
+  HRESULT Invoke(
+      [in] ICoreWebView2StagingFind* sender,
+      [in] IUnknown* args);
+}
+/// Receives `FindMatchCountChanged` events.
+[uuid(cecb8e8f-b6c8-55c3-98b1-68fd1e2b9eea), object, pointer_default(unique)]
+interface ICoreWebView2StagingFindMatchCountChangedEventHandler : IUnknown {
+  /// Provides the event args for the corresponding event.
+  HRESULT Invoke(
+      [in] ICoreWebView2StagingFind* sender,
+      [in] IUnknown* args);
+}
+
+
+/// Interface providing methods and properties for finding and navigating through text in the web view.
+/// This interface allows for finding text, navigation between matches, and customization of the find UI.
+// MSOWNERS: core (maxwellmyers@microsoft.com)
+[uuid(9c494a0a-c5d8-5fee-b7e6-4926d8d7b391), object, pointer_default(unique)]
+interface ICoreWebView2StagingFind : IUnknown {
+  /// Retrieves the index of the currently active match in the find session. Returns the index of the currently active match, or -1 if there is no active match.
+  // MSOWNERS: core (maxwellmyers@microsoft.com)
+  [propget] HRESULT ActiveMatchIndex([out, retval] UINT32* value);
+
+
+  /// Gets the total count of matches found in the current document based on the last find sessions criteria. Returns the total count of matches.
+  // MSOWNERS: core (maxwellmyers@microsoft.com)
+  [propget] HRESULT MatchCount([out, retval] UINT32* value);
+
+
+  /// Gets the `ShouldHighlightAllMatches` property.
+  // MSOWNERS: core (maxwellmyers@microsoft.com)
+  [propget] HRESULT ShouldHighlightAllMatches([out, retval] BOOL* value);
```

Comment:
No event for when this changes? How would the custom UI know when to update its checkbox?

Similarly, how does the custom UI know when to dismiss (when StopFind is called)? How does the custom UI know whether to show the "case sensitive" box checked or unchecked? It seems that the current API is not sufficient to build the custom Find UI.

Reply:
I think the custom UI is driving all of this, not responding to it. E.g. it's the custom UI that defines if matches should be highlighted, and then updates the WebView to do the right thing

Reply:
Correct. This property is driven entirely by the custom UI so no need for an event.